docs/flows.md

```diff
@@ -0,0 +1,311 @@
+
+parameters:
+  parameterName1:
+    description: optional parameter description
+    default: optional parameter default value
+  parameterName2:
+```
+
+Flow resource must have `Flow` kind and `appcontroller.k8s/v1alpha1` API version. However, despite looking similar
+to other Kubernetes resources, Flows are are not real resource. One cannot upload them to Kubernetes as is.
+Instead, they must be wrapped inside `Definition` resources. This is consistent with how other graph vertexes
+are represented. However, for other resource types it is still possible to create them in Kubernetes with
+`kubectl create -f resource.yaml`. But for flows it is always `cat flow.yaml | kubeac wrap | kubectl create -f-`.
+
+Sections below explain each of the flow properties and how they affect graph deployment.
+
+Note: Since it is possible to run `kubeac` both inside and outside of the cluster, here and below I refer to 
+`kubeac` binary by its simple name. However, in most cases this is going to be something like
+`kubectl exec k8s-appcontroller kubeac` to run the binary remotely, in AppController pod.
+
+### Flow name
+
+Each flow must have a `name` in its `metadata`. This name is used to refer to the flow in order to run it.
+There are two method to run a flow:
+1) Call it from withing a dependency graph. This is done by placing dependency on the `flow/flowName` resource,
+   where `flowName` is the name of the flow.
+2) Explicitly call the flow from command line: `kubeac run flowName`.
+
+The later method is only possible for exported flows. Exportable flows are the flows that have `exported: true`
+in their definition. Such flows are explicitly designed to be used by the user, rather than being of an internal use.
+
+When running flow from commandline, flow name may be omitted. In this case `DEFAULT` is used, which is the name of
+default flow - the main dependency graph.
+
+### Flow scope
+
+Since flow is a dependency graph on its own, having flow resource alone is not enough. There must be a way to
+identify, which resources belong to the flow. This is achieved by traversing the graph starting from the flow
+vertex. All vertices that are reachable from this vertex using dependencies with labels that match `construction`
+or `destruction` selectors of the flow are said to belong to the flow.
+
+Both `construction` and optional `destruction` fields of the `Flow` resource are dictionaries that state what
+labels dependencies must have in order for child resources of those dependencies to be included in the flow.
+
+Flows may consume other flows by making dependencies of the consumed flow resource. Moreover, that might be
+other resources in the consuming flow that depends on the flow, being consumed. In this case consumed flow is
+going to be a parent in both dependencies that form the flow and those, that are part of consuming flow. So it
+important to choose label selectors that do not overlap. If during graph traversal from a flow vertex AppController
```

Comment:
It is normal for selectors to overlap because DEFAULT flow has empty selector by-default (what else can it be?) that overlaps with any other selector